File: elovate_frontend/src/dtos/userProfileDto.ts
/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */
import {UserResponseDto} from "./userResponseDto";

export interface UserProfileDto extends UserResponseDto {

  friend: boolean;

  requested: boolean;

  requesting: boolean;

  self: boolean;

  blocked: boolean;

}

File: elovate_backend/src/main/java/com/tuwien/elovate/dtos/user/UserProfileDto.java
package com.tuwien.elovate.dtos.user;

/*-
 * #%L
 * ELOvate
 * %%
 * Copyright (C) 2024 ELOvate GmbH.
 * %%
 * Copyright (C) 2024 ELOvate GmbH. - All Rights Reserved
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * #L%
 */

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@AllArgsConstructor
@Getter
@Setter
@NoArgsConstructor
public class UserProfileDto extends UserResponseDto {

    private boolean isFriend;

    private boolean isRequested;

    private boolean isRequesting;

    private boolean isSelf;

    private boolean isBlocked;

    public UserProfileDto(UserResponseDto userResponseDto) {
        super(userResponseDto.getId(), userResponseDto.getEmail(), userResponseDto.getNickName());
    }
}
